src/curl.rs

```diff
@@ -17,8 +18,23 @@ where
         .arg(&url)
         .output()
         .map(move |result| match result {
-            Ok(output) => f(&output.stdout),
-            Err(err) => Err(Error::from(err)),
+            Ok(Output {
+                status,
+                stdout,
+                stderr,
+                ..
+            }) => match status.code() {
+                Some(0) => f(&stdout),
+                Some(n) => Err(Error::Custom(format!(
+                    "curl exited with non-0 status: {}. stderr: {:?}",
+                    n, stderr
+                ))),
+                None => Err(Error::Custom(format!(
+                    "curl terminated by signal. stderr: {:?}",
+                    stderr
+                ))),
+            },
+            Err(err) => Err(Error::Custom(format!("curl failed to execute: {:?}", err))),
```

Comment:
@madninja Not sure what exactly you meant by "more idiomatic rust case if expression", but I switched it to match on status code, which uncovered yet another failure variety - termination by signal.

Stringy errors _are_ kinda smelly, but IMO fine for errors you intend to just log rather than case-analyze. What would you suggest instead?

Reply:
Maybe define something like `Error::Cmd`?

Reply:
Since the curl module itself isn't the top level app, perhaps we define a this_error derived Error for curl errors? That contains the curl errors and makes them more contextual. Then have gateway-rs's Error enum enhanced with a `CurlError(curl::Error)`

Reply:
As for the “idiomatic match casing” I would lean toward a railway style ` if let` syntax for the success case and catch the failed exit code or failed execution in the `else` blocks